File: mobile/lib/core/utils/money.dart
import 'package:intl/intl.dart';

/// All monetary amounts from the backend are **integer paise** (₹1 = 100
/// paise). Format for display without ever doing float math on money.
class Money {
  const Money._();

  static final NumberFormat _inr = NumberFormat.currency(
    locale: 'en_IN',
    symbol: '₹',
    decimalDigits: 2,
  );

  /// `12345` (paise) → `₹123.45`.
  static String format(int paise) => _inr.format(paise / 100);

  /// `12300` (paise) → `₹123` (drops `.00`), else `₹123.45`.
  static String formatCompact(int paise) {
    final rupees = paise / 100;
    if (paise % 100 == 0) {
      return NumberFormat.currency(locale: 'en_IN', symbol: '₹', decimalDigits: 0)
          .format(rupees);
    }
    return _inr.format(rupees);
  }
}

File: frontend/lib/format/money.ts
/**
 * Money is stored and transported as integer paise (see backend PLAN §4).
 * Format at the UI edge only; never do arithmetic on the formatted string.
 */

const inr = new Intl.NumberFormat('en-IN', {
  style: 'currency',
  currency: 'INR',
  maximumFractionDigits: 0,
})

const inrWithPaise = new Intl.NumberFormat('en-IN', {
  style: 'currency',
  currency: 'INR',
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
})

/** paise → "₹1,234" (or "₹1,234.50" when there are non-zero paise). */
export function formatMoney(paise: number): string {
  const rupees = paise / 100
  return Number.isInteger(rupees) ? inr.format(rupees) : inrWithPaise.format(rupees)
}

/** paise → plain number of rupees (for chart values). */
export function toRupees(paise: number): number {
  return paise / 100
}
